Tests.Unit.Core.ApplicationServices/ApplicationServices/SystemUsage/ItSystemUsageWriteServiceTest.cs

```diff
@@ -845,6 +851,157 @@ private OrganizationUnit CreateOrganizationUnit(Guid uuid, Organization organiza
+            var createResult = _sut.Create(new SystemUsageCreationParameters(systemUuid, organizationUuid, input));
+
+            //Assert
+            Assert.True(createResult.Ok);
+            AssertTransactionCommitted(transactionMock);
+        }
+
+        [Fact]
+        public void Cannot_Create_With_Roles_If_Role_Assignment_Fails()
+        {
+            //Arrange
+            var (systemUuid, organizationUuid, transactionMock, organization, itSystem, itSystemUsage) = CreateBasicTestVariables();
+            var roleUuid = A<Guid>();
+            var userUuid = A<Guid>();
+
+            var input = CreateSystemUsageUpdateParametersWithData(roleUuid, userUuid);
+            var error = A<OperationFailure>();
+
+            SetupBasicCreateThenUpdatePrerequisites(organizationUuid, organization, systemUuid, itSystem, itSystemUsage);
+            ExpectRoleAssignmentReturns(itSystemUsage, roleUuid, userUuid, new OperationError(error));
+
+            //Act
+            var createResult = _sut.Create(new SystemUsageCreationParameters(systemUuid, organizationUuid, input));
+
+            //Assert
+            Assert.True(createResult.Failed);
+            AssertTransactionNotCommitted(transactionMock);
+            Assert.Equal(error, createResult.Error.FailureType);
+        }
+
+        [Fact]
+        public void Can_Update_Roles_To_Remove_Them()
+        {
+            //Arrange
+            var (systemUuid, organizationUuid, transactionMock, organization, itSystem, itSystemUsage) = CreateBasicTestVariables();
+            var roleUuid = A<Guid>();
+            var userUuid = A<Guid>();
+
+            var addRoleInput = CreateSystemUsageUpdateParametersWithData(roleUuid, userUuid);
+
+            var right = CreateRight(itSystemUsage, roleUuid, userUuid);
+
+            SetupBasicCreateThenUpdatePrerequisites(organizationUuid, organization, systemUuid, itSystem, itSystemUsage);
+            ExpectRoleAssignmentReturns(itSystemUsage, roleUuid, userUuid, right);
+
+            var createResult = _sut.Create(new SystemUsageCreationParameters(systemUuid, organizationUuid, addRoleInput));
+            Assert.True(createResult.Ok);
+            AssertTransactionCommitted(transactionMock);
+
+            var removeRoleInput = CreateSystemUsageUpdateParametersWithoutData();
+            ExpectRoleRemoveReturns(itSystemUsage, roleUuid, userUuid, right);
+            ExpectGetSystemUsageReturns(createResult.Value.Uuid, createResult.Value);
+
+            //Act
+            var updateResult = _sut.Update(createResult.Value.Uuid, removeRoleInput);
+
+            //Assert
+            Assert.True(updateResult.Ok);
+            AssertTransactionCommitted(transactionMock);
+        }
+        private static SystemUsageUpdateParameters CreateSystemUsageUpdateParametersWithoutData()
+        {
+            return new SystemUsageUpdateParameters()
+            {
+                Roles = Maybe<UpdatedSystemUsageRoles>.Some(new UpdatedSystemUsageRoles()
```

Comment:
Husk at du kan bruge FromNullable() extension i stedet for alle de der lange factory metoder